container-search/src/main/java/com/yahoo/search/searchers/ValidateNearestNeighborSearcher.java

```diff
@@ -0,0 +1,127 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.search.searchers;
+
+import com.google.common.annotations.Beta;
+
+import com.yahoo.container.QrSearchersConfig;
+import com.yahoo.prelude.query.Item;
+import com.yahoo.prelude.query.NearestNeighborItem;
+import com.yahoo.prelude.query.ToolBox;
+import com.yahoo.search.Query;
+import com.yahoo.search.Result;
+import com.yahoo.search.Searcher;
+import com.yahoo.search.query.ranking.RankProperties;
+import com.yahoo.search.result.ErrorMessage;
+import com.yahoo.search.searchchain.Execution;
+import com.yahoo.tensor.Tensor;
+import com.yahoo.tensor.TensorType;
+import com.yahoo.vespa.config.search.AttributesConfig;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Validates any NearestNeighborItem query items.
+ *
+ * @author arnej
+ */
+@Beta
+public class ValidateNearestNeighborSearcher extends Searcher {
```

Comment:
NearestNeighborValidatingSearcher? Or just NearestNeighborValidator?

Reply:
searcher name was based on existing searchers:
ValidateMatchPhaseSearcher.java
ValidateSortingSearcher.java
ValidatePredicateSearcher.java

Reply:
I no like those names, but yes consistency is better.